picard/util/__init__.py

```diff
@@ -1183,4 +1184,10 @@ def detect_unicode_encoding(path):
         for bom, encoding in ENCODING_BOMS.items():
             if first_bytes.startswith(bom):
                 return encoding
-        return 'utf-8'
+        rawdata = f.read()
+        result = chardet.detect(rawdata)
```

Comment:
Should we do `first_bytes + rawdata` here, just in case that we don't muss probably relevant bytes?